### ISSUE TITLE
IE7 "Permission Denied" error

### ISSUE THREAD

**Issue by amoslanka** [asual/jquery-address#48]
I'm getting an error on page load when testing IE7 that seems to be coming from line 213 of jquery address. The error is not thrown if I don't load the js file, and is otherwise hard to nail down any specifics because of IE7's general difficulty in debugging. 

The error is called "Permission denied", coming from "var win = _frame.contentWindow;" in:

```
_st(function() {
    $(_frame).bind('load',
    function() {
        var win = _frame.contentWindow;
        _value = win[ID] !== UNDEFINED ? win[ID] : '';
        if (_value != _href()) {
            _update(FALSE);
            _l.hash = _crawl(_value, TRUE);
        }
    });
    if (_frame.contentWindow[ID] === UNDEFINED) {
        _html();
    }
},
50);
```

This block of code is called only on IE6 and IE7. "if (_msie && _version < 8)"

Inserting an alert(win[ID]); will throw the error if placed ahead of the line where win[ID] is first referenced. 

I'm not too familiar with how to deal with IE7 issues, does anyone else out there experience this error? My thought is that it might have to do with accessing properties of another frame (sandbox issues) though of course unless i'm mistaken, the frame that is created for history purposes shouldn't be outside the sandbox.

Any help would be greatly appreciated!

**Comment by lukeshumard** [comment]
what is in your response from the load?

**Comment by amoslanka** [comment]
New information. I've had a hell of a time getting the error to print out in a way that I can follow it, the way firebug or web inspector do. Finally found Companion.js, which gives me a console.log in IE7. 

Seems clear to me now that the error was referring to actions that are impermissable to carry out on an iframe outside the sandbox. Didn't make sense at the time, but as it turns out there seems to be something in IE7 that was putting it out of the sandbox when it goes to write the document.domain property of the iframe. 

The thing that seems to have brought out this issue and complicated things is that I'm testing through a domain using a port (rails uses port 3000). Of course it seems that most dev work and testing is done on localhost or something of the like, so finding one circumstance on a specific browser through a domain that is less common would lead to a strange error like this not getting caught. 

Anyway, I tracked the issue down to line 140 (jquery.address-1.3.1.js) where the _html method is writing to string the src that will become the iframe used for the history. 

```
            var src = _js() + ':' + FALSE + ';document.open();document.writeln(\'<html><head><title>' + 
                _d.title.replace('\'', '\\\'') + '</title><script>var ' + ID + ' = "' + encodeURIComponent(_href()) + 
                (_d.domain != _l.host ? '";document.domain="' + _d.domain : '') + 
                '";</' + 'script></head></html>\');document.close();';
```

On line 140 the iframe writes its own document.domain _if the current document's domain is different from its location's host._ If you are serving pages using a port, http://localhost:3000, for example, this comparison results in a false, because _d.domain is _localhost_ and _l.host includes the port, _localhost:3000_. This results in the line setting document.domain being written to the iframe, leading to my "Permission Denied" error because somewhere in there, apparently, IE7 is interpreting the iframe's domain as different from the root document.

Now I'm no sure about this 100%, it seems that enforcing a domain into the iframe should result in its being permissable. But of course it seems to not be an issue of the port needing to be included in the iframe's domain because giving it a domain with a port results in an invalid argument error. I'm not sure if I'm missing something in the wild world of browsers (especially IE browsers) or the core functionality of iframes that works in ways I'm not aware of. 

Either way, it seems to me that this comparison is likely suppose to be a "port-less" comparison. I changed the comparison line from 

```
_d.domain != _l.host
```

to

```
_d.domain != _l.hostname
```

which results in the document.domain=... line being left out of the written src. 

This seems to have worked for me, and there is no longer a sandbox issue. Unless I'm unaware of something else, it seems that the _l.hostname comparison is what was intended, and there should be no document.domain assignment line in the iframe, at least serving to IE7 (and possibly older versions of IE) through a url including a port.
